traitsui/editors/api.py

```diff
@@ -7,11 +7,15 @@
 try:
     from .array_editor import ArrayEditor
 except ImportError as e:
-    if 'numpy' not in e.message:
```

Comment:
Two, three, even four `try`s would be perfectly fine, IMO. This is done exactly once. It can afford to be very mildly complicated.